cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -187,21 +192,26 @@ fn register_weth_token_from_ethereum_to_asset_hub() {
 	// Fund AssetHub sovereign account so that it can pay execution fees.
 	BridgeHubRococo::fund_para_sovereign(AssetHubRococo::para_id().into(), INITIAL_FUND);
 
-	let message_id: H256 = [1; 32].into();
-
 	BridgeHubRococo::execute_with(|| {
 		type RuntimeEvent = <BridgeHubRococo as Chain>::RuntimeEvent;
+		type RuntimeOrigin = <BridgeHubRococo as Chain>::RuntimeOrigin;
 		type EthereumInboundQueue =
 			<BridgeHubRococo as BridgeHubRococoPallet>::EthereumInboundQueue;
-		let message = VersionedMessage::V1(MessageV1 {
-			chain_id: CHAIN_ID,
-			command: Command::RegisterToken { token: WETH.into(), fee: XCM_FEE },
-		});
-		let (xcm, fee) = EthereumInboundQueue::do_convert(message_id, message).unwrap();
 
-		assert_ok!(EthereumInboundQueue::burn_fees(AssetHubRococo::para_id().into(), fee));
+		let register_asset_message = make_register_asset_message();
 
-		let _ = EthereumInboundQueue::send_xcm(xcm, AssetHubRococo::para_id().into()).unwrap();
+		EthereumBeaconClient::store_execution_header(
+			register_asset_message.message.proof.block_hash,
+			register_asset_message.execution_header,
+			0,
+			H256::default(),
+		);
+
+		EthereumInboundQueue::submit(
+			RuntimeOrigin::signed(BridgeHubRococoSender::get()),
+			register_asset_message.message,
+		)
+		.unwrap();
```

Comment:
Hmmm I'm not sure this change is a good one. What do you think is the benefit of changing this? I feel like the previous way is clearer to read and understand what is happening, imo.

Reply:
To not expose the internal implementation details to the tests. 
More context in https://github.com/Snowfork/snowbridge/pull/1115#discussion_r1458758342